go-controller/pkg/libovsdbops/model_client.go

```diff
@@ -385,53 +389,76 @@ func (m *modelClient) Lookup(opModels ...operationModel) error {
 	return err
 }
 
+// CreateOrUpdate, Delete and Lookup can be called to
+// 1. create or update a single model
+// Model should be set, bulkOp = false, errNotfound = false
+// 2. update/delete/lookup 0..n models (create can't be done for multiple models at the same time)
+// Model index or predicate should be set
+//
+// The allowed combination of operationModel fields is different for these cases.
+// Both Model db index, and ModelPredicate can only be empty for the first case
+func lookupRequired(opModel *operationModel) bool {
+	// we know create is not supposed to be performed, if these fields are set
+	if opModel.BulkOp || opModel.ErrNotFound {
+		return true
+	}
+	return false
+}
+
 // lookup the model in the cache prioritizing provided indexes over a
-// predicate unless bulkop is set
+// predicate
+// If lookup was successful, opModel.Model will have UUID set,
+// so that further user operations with the same model are indexed by UUID
 func (m *modelClient) lookup(opModel *operationModel) error {
 	if opModel.ExistingResult == nil && opModel.Model != nil {
 		opModel.ExistingResult = getListFromModel(opModel.Model)
 	}
 
 	var err error
-	if opModel.Model != nil && !opModel.BulkOp {
-		err = m.get(opModel)
-		if err == nil || err != client.ErrNotFound {
+	if opModel.Model != nil {
+		err = m.where(opModel)
+		if err != errNoIndices {
+			// if index wasn't provided by the Model, try predicate search
+			// otherwise return where result
 			return err
 		}
 	}
-
+	// if index wasn't provided by the Model (errNoIndices) or Model == nil, try predicate search
 	if opModel.ModelPredicate != nil {
-		err = m.whereCache(opModel)
-	} else if opModel.BulkOp {
-		panic("Expected a ModelPredicate with BulkOp==true")
+		return m.whereCache(opModel)
 	}
-
-	return err
+	// the only operation that can be performed without a lookup (it can have no db indices and no ModelPredicate set)
+	// is Create.
+	if lookupRequired(opModel) {
+		return fmt.Errorf("provided operationModel should have either Model db index, or ModelPredicate set: " +
+			"lookup can't be performed")
```

Comment:
I would rephrase to something like
`missing model indixes or predicate when a lookup was required`